torch/onnx/symbolic_opset9.py

```diff
@@ -722,16 +728,44 @@ def prelu(g, self, weight):
 def silu(g, input):
     return g.op("Mul", input, g.op("Sigmoid", input))
 
+
 def mish(g, input):
     return g.op("Mul", input, g.op("Tanh", g.op("Softplus", input)))
 
+
+def op_with_optional_float_cast(g, op_name, self, *args, other_operands=None, opset_before=None, target_float_t="Float", **kwargs):
+    origin_dtype = self.type().scalarType()
+    require_cast = not sym_help._is_fp(self) and (opset_before is None or sym_help._export_onnx_opset_version < opset_before)
+    if other_operands is None:
+        other_operands = []
+
+    if require_cast:
+        warnings.warn(
+            f"{op_name}-{sym_help._export_onnx_opset_version} only support float types."
+            f"Using cast<{target_float_t}>-{op_name}-cast<{origin_dtype}> instead.")
+        self = g.op("Cast", self, to_i=sym_help.cast_pytorch_to_onnx[target_float_t])
+        for i in range(len(other_operands)):
+            if not sym_help._is_fp(other_operands[i]):
+                other_operands[i] = g.op("Cast", other_operands[i], to_i=sym_help.cast_pytorch_to_onnx[target_float_t])
+
+    self = g.op(op_name, self, *other_operands, *args, **kwargs)
+
+    if require_cast:
+        self = g.op(
```

Comment:
why are we wrapping the output with a cast?

Reply:
Because we are using, say `Cast<int>Clip<float>(Cast<float>(int_tensor))` to mimic `Clip<int>(int_tensor)`. The return type should match the original type. But of course, to be more rigorous, we'd better infer the original output types rather than simply assuming that output type = input type; however, it is adequate for the mentioned operators. :-)